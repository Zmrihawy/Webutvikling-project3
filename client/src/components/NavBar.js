import React from "react";
import PropTypes from "prop-types";
import TypoGraphy from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";

function NavBar(props) {
  const { loggedInUser } = props;
  return (
    <React.Fragment>
      <Link to="/" style={{ textDecoration: "none" }}>
        <Button label="Home" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Home
          </TypoGraphy>
        </Button>
      </Link>

      <Link to="/component-list" style={{ textDecoration: "none" }}>
        <Button label="component-list" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Component list
          </TypoGraphy>
        </Button>
      </Link>

      <Link to="/categories" style={{ textDecoration: "none" }}>
        <Button label="categories" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Categories
          </TypoGraphy>
        </Button>
      </Link>

      <Link to="/shopping-cart" style={{ textDecoration: "none" }}>
        <Button label="shopping-cart" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Shopping cart
          </TypoGraphy>
        </Button>
      </Link>

      <Link to="/user" style={{ textDecoration: "none" }}>
        <Button label="User" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            User
          </TypoGraphy>
        </Button>
      </Link>

      <div style={{ flex: 1 }} />

      <TypoGraphy justify="space-between">
        {loggedInUser && loggedInUser.username
          ? "Logged in as " + props.loggedInUser.username
          : "Not logged in"}
      </TypoGraphy>
    </React.Fragment>
  );
}

NavBar.propTypes = {
  loggedInUser: PropTypes.shape({
    username: PropTypes.string,
    shoppingBasket: PropTypes.array
  })
};

export default NavBar;
