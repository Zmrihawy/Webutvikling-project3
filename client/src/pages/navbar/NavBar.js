import React from "react";
import PropTypes from "prop-types";
import TypoGraphy from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";


// The NavBar component which is used in MainBar
function NavBar(props) {
  const { loggedInUser } = props;
  return (
    <React.Fragment>
      {/**Link to HomePage */}
      <Link to="/" style={{ textDecoration: "none" }}>
        <Button label="Home" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Home
          </TypoGraphy>
        </Button>
      </Link>
       {/**view List of components */}
      <Link to="/component-list" style={{ textDecoration: "none" }}>
        <Button label="component-list" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Browse
          </TypoGraphy>
        </Button>
      </Link>
       {/**Link to shopping cart */}
      <Link to="/shopping-cart" style={{ textDecoration: "none" }}>
        <Button label="shopping-cart" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Cart
          </TypoGraphy>
        </Button>
      </Link>
      {/**To user page where the user can creat new user or can logg in  */}
      <Link to="/user" style={{ textDecoration: "none" }}>
        <Button label="User" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            User
          </TypoGraphy>
        </Button>
      </Link>
      {/** visualization of or data */}
      <Link to="/tag-cloud" style={{ textDecoration: "none" }}>
        <Button label="Tag Cloud" style={{ textTransform: "none" }}>
          <TypoGraphy variant="subtitle1" style={{ color: "white" }}>
            Visualization
          </TypoGraphy>
        </Button>
      </Link>

      <div style={{ flex: 1 }} />
          {/** show how is logged in */}
      <TypoGraphy justify="space-between">
        {loggedInUser && loggedInUser.username
          ? "Logged in as " + props.loggedInUser.username
          : "Not logged in"}
      </TypoGraphy>
    </React.Fragment>
  );
}

NavBar.propTypes = {
  loggedInUser: PropTypes.shape({
    username: PropTypes.string,
    shoppingCart: PropTypes.array
  })
};

export default NavBar;
