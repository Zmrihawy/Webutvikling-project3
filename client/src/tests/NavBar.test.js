import React from "react";
import NavBar from "../pages/navbar/NavBar"
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

it("renders tag search bar", () => {
  let loggedInUser = {
    username: 'foo',
    shoppingCart: []
  }
  const tree = renderer.
    create(
      <Router>
        <NavBar loggedInUser={loggedInUser} />
      </Router>
    )

  expect(tree).toMatchSnapshot();
});

