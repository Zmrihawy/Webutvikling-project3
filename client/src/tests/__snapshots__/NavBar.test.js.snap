// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders tag search bar 1`] = `
Array [
  <a
    href="/"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
      disabled={false}
      label="Home"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Home
        </h6>
      </span>
    </button>
  </a>,
  <a
    href="/component-list"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
      disabled={false}
      label="component-list"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Browse
        </h6>
      </span>
    </button>
  </a>,
  <a
    href="/shopping-cart"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
      disabled={false}
      label="shopping-cart"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Cart
        </h6>
      </span>
    </button>
  </a>,
  <a
    href="/user"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
      disabled={false}
      label="User"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "white",
            }
          }
        >
          User
        </h6>
      </span>
    </button>
  </a>,
  <a
    href="/tag-cloud"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-text"
      disabled={false}
      label="Tag Cloud"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "textTransform": "none",
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Visualization
        </h6>
      </span>
    </button>
  </a>,
  <div
    style={
      Object {
        "flex": 1,
      }
    }
  />,
  <p
    className="MuiTypography-root MuiTypography-body1"
    justify="space-between"
  >
    Logged in as foo
  </p>,
]
`;
